cumulusci/utils/yaml/cumulusci_yml.py

```diff
@@ -60,120 +180,731 @@ def _check(cls, values):
+        description="The dependency resolution configuration for the project. See https://cumulusci.readthedocs.io/en/stable/dev.html#manage-dependencies for more information.",
+        examples=[
+            {
+                "production": "latest_release",
+                "preproduction": "include_beta",
+                "resolution_strategies": {
+                    "latest_release": ["tag", "latest_release", "unmanaged"],
+                    "include_beta": [
+                        "tag",
+                        "latest_beta",
+                        "latest_release",
+                        "unmanaged",
+                    ],
+                },
+            }
+        ],
+    )
+    dependency_pins: Optional[List[Dict[str, str]]] = Field(
+        None,
+        title="Dependency Pins",
+        description="A list of dependency pins for the project. See https://cumulusci.readthedocs.io/en/stable/dev.html#manage-dependencies for more information.",
+        examples=[
+            [
+                {
+                    "github": "https://github.com/SalesforceFoundation/NPSP",
+                    "tag": "rel/3.219",
+                },
+                {
+                    "github": "https://github.com/SalesforceFoundation/Contacts_and_Organizations",
+                    "tag": "rel/3.19",
+                },
+            ],
+        ],
+    )
+    source_format: Literal["sfdx", "mdapi"] = Field(
+        "mdapi",
+        title="Source Format",
+        description="The source format for the project. Either sfdx for sfdx source format under force-app/ or mdapi for Metadata API format under src/.",
+        examples=["sfdx", "mdapi"],
+    )
+    custom: Optional[Dict] = Field(
+        None,
+        title="Custom",
+        description="A dictionary for custom project configuration. This is a good place to put project-specific configuration. Do not use for storing any sensitive information like credentials!",
+        examples=[{"my_custom_key": "my_custom_value"}],
+    )
+
+    class Config:
+        title = "Project Configuration"
+        description = "The project configuration for the project. See https://cumulusci.readthedocs.io/en/stable/config.html for more information."
 
 
 class ScratchOrg(CCIDictModel):
-    config_file: Path = None
-    days: int = None
-    namespaced: str = None
-    setup_flow: str = None
-    noancestors: bool = None
-    release: Literal["preview", "previous"] = None
+    config_file: Path = Field(
+        None,
+        title="Config File",
+        description="The path to the scratchdef config file.",
+        examples=[
+            "orgs/dev.json",
+            "orgs/managed.json",
+            "config/project-scratch-def.json",
+        ],
+    )
+    days: int = Field(
+        None,
+        title="Days",
+        description="The number of days the scratch org should last.",
+        min=1,
+        max=30,
+        examples=[1, 7, 30],
+    )
+    namespaced: bool = Field(
```

Comment:
yay! the previous definition was wrong (claimed this was a str) and so YAML checking was failing in VSCode